app/routes/submit.js

```diff
@@ -35,14 +41,21 @@ export default Ember.Route.extend(Analytics, ResetScrollMixin, CasAuthenticatedR
         this.setupSubmitController(controller, model);
         return this._super(...arguments);
     },
-    actions: {
-        willTransition: function(transition) {
-            // Displays confirmation message if user attempts to navigate away from Add Preprint process with unsaved changes
-            var hasFile = this.controller.get('file') !== null || this.controller.get('selectedFile') !== null;
+    isPageDirty() {
+        // If true, shows a confirmation message when leaving the page
+        // True if the user already created/chosen a project node
+        return this.controller.get('hasDirtyFields');
+    },
+    shouldCheckIsPageDirty(transition) {
```

Comment:
This method is confusingly named. One would think is should return false to skip the isPageDirty check, but you actually return true (as you have done and the docs indicate). They should have named this `shouldSkipIsPageDirty`.